Flexible MySQL: Adding apis for BackupAndExport/LTR Backup

rishky1001 opened the pull request:
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

rishky1001 committed:
Flexible MySQL: Adding apis for BackupAndExport/LTR Backup

rishky1001 committed:
Fixing model validation errors

rishky1001 committed:
Fixing lint errors

rishky1001 committed:
Fixing swagger api version

rishky1001 committed:
Prettier fix

rishky1001 committed:
Fixing lint check

rishky1001 committed:
Adding existing apis in readme for new tag

rishky1001 committed:
Changing prebackup response parameter and adding base class for storage details.

rishky-msft commented:
@tianxchen-ms 
We are getting PreCheck duplicate schema error on the common-type items. 
https://github.com/Azure/azure-rest-api-specs/pull/20675/checks?check_run_id=8330448076

We have used the same way for previous api versions also.
Can you help us if this need to be fixed?
If yes, how can we fix this?

rkmanda commented on specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-08-01-privatepreview/BackupAndExport.json:
```diff
@@ -0,0 +1,338 @@
+    "title": "MySQLManagementClient",
+    "description": "The Microsoft Azure management API provides create, read, update, and delete functionality for Azure MySQL resources including servers, databases, firewall rules and configurations.",
+    "version": "2022-08-01-privatepreview"
+  },
```
move to the top of the file

sumitkumams replied:
done

rkmanda commented on specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-08-01-privatepreview/BackupAndExport.json:
```diff
@@ -0,0 +1,338 @@
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  }
```
move to the top of the file

sumitkumams replied:
done

rkmanda commented on specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-08-01-privatepreview/BackupAndExport.json:
```diff
@@ -0,0 +1,338 @@
+  "consumes": [
+    "application/json"
+  ],
+  "definitions": {
```
>definitions [](http://example.com/codeflow?start=3&length=11)

move definitions after paths

sumitkumams replied:
done

rkmanda commented on specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-08-01-privatepreview/BackupAndExport.json:
```diff
@@ -0,0 +1,338 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/flexibleServers/{serverName}/preBackup": {
```
>preBackup [](http://example.com/codeflow?start=132&length=9)

validateBackup?

sumitkumams replied:
modified to validateBackup

sumitkumams committed:
changes after PR comments

1. change name from "PreBackup" to "ValidateBackup"
2. add tag in backupFormat "INTERNAL - DO NOT USE"
3. change the order of securitydefinitions  , title, paths

sumitkumams committed:
updated description for "CollatedFormat" do not use.

just added description for "CollatedFormat" to not use.
